legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/gitlab/auth/GitLabTokenManager.java

```diff
@@ -107,6 +118,10 @@ void putAccessToken(GitLabMode mode, String token)
         {
             throw new IllegalArgumentException("token may not be null");
         }
+        if (!token.getTokenType().equals(TokenType.OAUTH2_ACCESS) && !token.getTokenType().equals(TokenType.PRIVATE))
+        {
+            throw new UnsupportedOperationException(token.getTokenType().toString() + " token is not supported");
+        }
```

Comment:
I think a not null check is sufficient here. We're not really making use of the type, other than to pass it to the constructor of GitLabApi.